### ISSUE TITLE
multibootusb 9.2 - Manjaro - Start kernel first.

### ISSUE THREAD

**Issue by nobicycle** [mbusb/multibootusb#329]
Problem 1.
When booting Manjaro 17 KDE a the splash screen appears but just hangs

Problem 2.
Choose Grub boot, the following is reported:
Cannot find vmlinux
...
Start kernel first

Regards
Thanks for the app.

**Comment by nobicycle** [comment]
> [me@tower multibootusb]$ find . -name grub.cfg|cat -n
>      1	./grub/grub.cfg
Attaching 1.
[1.grub.cfg.txt](https://github.com/mbusb/multibootusb/files/1887439/1.grub.cfg.txt)

[me@tower multibootusb]$ find . -name syslinux.cfg|cat -n
     1	./syslinux.cfg
     2	./archlabs-2018-03/arch/boot/syslinux/syslinux.cfg
     3	./archlabs-2018-03/isolinux/syslinux.cfg
     4	./ubuntu-17.04-desktop-amd64/isolinux/syslinux.cfg
     5	./bl-Deuterium-amd64_20170429/isolinux/syslinux.cfg
     6	./Fatdog64-720/syslinux.cfg
     7	./manjaro-kde-17.0-stable-x86_64/isolinux/syslinux.cfg
     8	./manjaro-kde-17.0-stable-x86_64/manjaro/boot/syslinux/syslinux.cfg

Attaching 1., 7. and 8.
[1.syslinux.cfg.txt](https://github.com/mbusb/multibootusb/files/1887440/1.syslinux.cfg.txt)
[7.syslinux.cfg.txt](https://github.com/mbusb/multibootusb/files/1887441/7.syslinux.cfg.txt)
[8.syslinux.cfg.txt](https://github.com/mbusb/multibootusb/files/1887442/8.syslinux.cfg.txt)

**Comment by shinji-s** [comment]
Thank you for the logs.
When no grub config file is provided by a distro, mbusb tries to generate
one from syslinux config files. The problem is that this iso file carries 
syslinux config files for both i686 and x86_64 architecture and mbusb
is not (yet) smart enough to choose the correct one.
If you modify 'start' entry in /multibootusb/manjaro-kde-17.0-stable-x86_64/loopback.cfg
so that it looks like below, you should be able to boot. (Just replace 'i686' with 'x86_64'.)
```
menuentry "start" {
    linux /multibootusb/manjaro-kde-17.0-stable-x86_64/manjaro/boot/x86_64/vmlinuz misobasedir=/multibootusb/manjaro-kde-17.0-stable-x86_64/manjaro misodevice=/dev/disk/by-uuid/273D-A0F7 nouveau.modeset=1 i915.modeset=1 radeon.modeset=1 logo.nologo  showopts
    initrd /multibootusb/manjaro-kde-17.0-stable-x86_64/manjaro/boot/intel_ucode.img /multibootusb/manjaro-kde-17.0-stable-x86_64/manjaro/boot/x86_64/initramfs.img
}
```
I'm looking at syslinux booting issue. I suspect syslinux loader/module incompability
but it is quite difficult to diagnose when booting just hangs without any messages.

If you use  a Majaro 17.1.7 you should not have to do the above tweaking. It seems
they ditched syslinux for grub.

**Comment by shinji-s** [comment]
OK the culprit for syslinux boot hanging is again gfxboot module.
If you comment out the line

> UI boot/syslinux/gfxboot.c32 boot/syslinux/bootlogo boot/syslinux/syslinux.msg

in /manjaro-kde-17.0-stable-x86_64/manjaro/boot/syslinux/miso_head.cfg
then you can boot via syslinux. Although you lose the fancy logo.

**Comment by nobicycle** [comment]
Thank you

**Comment by nobicycle** [comment]
Hi, I just tried  manjaro-gnome-17.1.7-stable-x86_64.iso without tweaking and a 'corrupt  kerne'l was reported, even though the md5sum is correct and even though the iso runs under qemu.

I attach files.
[grub.cfg.txt](https://github.com/mbusb/multibootusb/files/1897090/grub.cfg.txt)
[gnome-grub.cfg.txt](https://github.com/mbusb/multibootusb/files/1897091/gnome-grub.cfg.txt)
[gnome-syslinux.cfg.txt](https://github.com/mbusb/multibootusb/files/1897092/gnome-syslinux.cfg.txt)

I did not try multibootusb grub menu option yet. I just selected Manjaro Gnome from the main menu.

**Comment by shinji-s** [comment]
Support for syslinux booting is not available anymore in 17.1.7.
You'll have to boot via grub.

**Comment by edward-p** [comment]
same issue on archlinux

**Comment by edward-p** [comment]
@mbusb Yes, after doing some similar grub fix, I can boot into it. Besides, Manjaro is based on Arch, they're quite the same 😀.

**Comment by Dinir** [comment]
I have the same problem installing Arch, and I don't get what similar grub fix I am supposed to do.
I'm attaching {target-drive}/syslinux.cfg and {target-drive}/grub/grub.cfg here.

[syslinux.cfg.txt](https://github.com/mbusb/multibootusb/files/2213889/syslinux.cfg.txt)
[grub.cfg.txt](https://github.com/mbusb/multibootusb/files/2213890/grub.cfg.txt)

**Comment by shinji-s** [comment]
I guess your target for the tweak is in /multibootusb/archlinux-2018.07.01-x86_64/loopback.cfg.

**Comment by Dinir** [comment]
I had to fix my computer by making a common usb boot disk - that is just having one iso wrote into the entire usb stick, so I think the cfg files I posted might not be relevant.

But I tried making a new one using multibootusb from a different computer and I am having the same problem, so posting the current `loopback.cfg` here.
[loopback.cfg.txt](https://github.com/mbusb/multibootusb/files/2268681/loopback.cfg.txt)

**Comment by shinji-s** [comment]
Please see https://github.com/mbusb/multibootusb/issues/405.
